azure_variables.tf

```diff
@@ -0,0 +1,82 @@
+variable "azure_subscription_id" {
+  type        = string
+  default     = null
+  description = "Azure subscription ID"
+}
+
+variable "azure_tenant_id" {
+  type        = string
+  default     = null
+  description = "Azure tenant ID"
+}
+
+variable "azure_client_id" {
+  type        = string
+  default     = null
+  description = "Client ID for Azure Service Principal. WARNING - This will be written to the state file in plain text"
+}
+
+variable "azure_client_secret" {
+  type        = string
+  default     = null
+  description = "Client secret for Azure Service Principal. WARNING - This will be written to the state file in plain text"
+}
+
+variable "azure_environment" {
+  type        = string
+  default     = "AzurePublicCloud"
+  description = "The Azure cloud environment. Valid values: AzurePublicCloud, AzureUSGovernmentCloud, AzureChinaCloud, AzureGermanCloud.  Defaults to AzurePublicCloud"
```

Comment:
You could omit this from the description as the `can` check will inform module users of a transgression and provide values

Reply:
good point

Reply:
just pushed the change @ksatirli